api/src/main/java/io/opentelemetry/common/ImmutableKeyValuePairs.java

```diff
@@ -128,4 +128,22 @@ private static void swap(Object[] data, int a, int b) {
     data[b] = keyA;
     data[b + 1] = valueA;
   }
+
+  @Override
+  public String toString() {
+    final StringBuilder sb = new StringBuilder("{");
+    forEach(
+        new KeyValueConsumer<V>() {
+          @Override
+          public void consume(String key, V value) {
+            sb.append(key).append(":").append(value).append(",");
+          }
+        });
+    // get rid of that last pesky comma
+    if (sb.length() > 1) {
+      sb.deleteCharAt(sb.length() - 1);
```

Comment:
Usually see `sb.setLength(sb.length() - 1)` for dealing with pesky commas. Checked `deleteCharAt` and it doesn't seem to avoid `System.arrayCopy` even for this simple case

Reply:
oh! good tip. I hadn't looked that deeply into the implementation of `deleteCharAt`. Probably not a big deal for a debugging toString method, but no reason not to change it, now that I know!